pkg/asset/manifests/dns.go

```diff
@@ -115,7 +116,10 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 			if err != nil {
 				return errors.Wrapf(err, "failed to get public zone for %q", installConfig.Config.BaseDomain)
 			}
-			config.Spec.PublicZone = &configv1.DNSZone{ID: zone.Name}
+			if installConfig.Config.GCP.PublicZoneID == "" {
+				installConfig.Config.GCP.SetPublicZoneID(strconv.FormatUint(zone.Id, 10))
+			}
+			config.Spec.PublicZone = &configv1.DNSZone{ID: installConfig.Config.GCP.PublicZoneID}
```

Comment:
Also i think you are setting the wrong ID for the zone. look at the docs https://github.com/openshift/api/blob/c99157bc6492218400b18cde7f66510f9ee56a9a/config/v1/types_dns.go#L55-L65 to see how the ID needs to be specified.

Reply:
The ID is correct. If you look at the documentation you point at, it will specify a link to the google API, where it is stated you can use either the managed zone name or the id. I can verify this works.